### ISSUE TITLE
All devices seem inaccessible in Tailnet after removing an ACL tag

### ISSUE THREAD

**Issue by joelheaps** [tailscale/tailscale#6197]
### What is the issue?

All Tailnet devices unable to communicate after attempting to rename ACL tag.

**Expected**: same traffic flow rules, with a renamed tag

**Experienced:**
- MagicDNS resolution, pinging other devices by tailscale IP fails
- Running `tailscale status` on connected devices only shows local device
- Primary workstation (macOS) shows connected in tray icon, but web interface shows offline, last seen at current time

Devices remain inaccessible even after reverting ACL file to a simple accept all rule.

#### Additional info and troubleshooting steps

Attempting service restart on one machine:

```shell
joel@flightdeck:~$ sudo systemctl restart tailscaled
joel@flightdeck:~$ sudo systemctl status tailscaled.service 
● tailscaled.service - Tailscale node agent
     Loaded: loaded (/lib/systemd/system/tailscaled.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2022-11-04 10:15:59 CDT; 8s ago
       Docs: https://tailscale.com/kb/
    Process: 55312 ExecStartPre=/usr/sbin/tailscaled --cleanup (code=exited, status=0/SUCCESS)
   Main PID: 55318 (tailscaled)
     Status: "Connected; flightdeck.taild7e4.ts.net; 100.81.78.19 fd7a:115c:a1e0:ab12:4843:cd96:6251:4e13"
      Tasks: 9 (limit: 2333)
     Memory: 11.0M
        CPU: 213ms
     CGroup: /system.slice/tailscaled.service
             └─55318 /usr/sbin/tailscaled --state=/var/lib/tailscale/tailscaled.state --socket=/run/tailscale/tailscaled.sock --port=41641

Nov 04 10:16:01 flightdeck tailscaled[55318]: magicsock: home is now derp-12 (ord)
Nov 04 10:16:01 flightdeck tailscaled[55318]: magicsock: endpoints changed: 24.252.3.19:54294 (stun), [2600:8804:1202:4c02:d880:ccff:febd:15cf]:41641 (stun), 10.1.10.137:41641 (local)
Nov 04 10:16:01 flightdeck tailscaled[55318]: magicsock: adding connection to derp-12 for home-keep-alive
Nov 04 10:16:01 flightdeck tailscaled[55318]: magicsock: 1 active derp conns: derp-12=cr0s,wr0s
Nov 04 10:16:01 flightdeck tailscaled[55318]: Switching ipn state Starting -> Running (WantRunning=true, nm=true)
Nov 04 10:16:01 flightdeck tailscaled[55318]: control: NetInfo: NetInfo{varies=false hairpin=false ipv6=true udp=true icmpv4=false derp=#12 portmap= link=""}
Nov 04 10:16:01 flightdeck tailscaled[55318]: derphttp.Client.Connect: connecting to derp-12 (ord)
Nov 04 10:16:01 flightdeck tailscaled[55318]: magicsock: derp-12 connected; connGen=1
Nov 04 10:16:01 flightdeck tailscaled[55318]: health("overall"): ok
Nov 04 10:16:01 flightdeck tailscaled[55318]: control: netmap: got new dial plan from control
joel@flightdeck:~$ tailscale status
100.81.78.19    flightdeck           flightdeck.taild7e4.ts.net linux   -
```

Attempting service restart on another machine:

```shell
root@oma-vs1:~# systemctl restart tailscaled
root@oma-vs1:~# systemctl status tailscaled
● tailscaled.service - Tailscale node agent
     Loaded: loaded (/lib/systemd/system/tailscaled.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2022-11-04 10:18:07 CDT; 5s ago
       Docs: https://tailscale.com/kb/
    Process: 2560742 ExecStartPre=/usr/sbin/tailscaled --cleanup (code=exited, status=0/SUCCESS)
   Main PID: 2560749 (tailscaled)
     Status: "Connected; oma-vs1.taild7e4.ts.net; 100.99.26.120 fd7a:115c:a1e0:ab12:4843:cd96:6263:1a78"
      Tasks: 11 (limit: 38335)
     Memory: 11.8M
        CPU: 197ms
     CGroup: /system.slice/tailscaled.service
             └─2560749 /usr/sbin/tailscaled --state=/var/lib/tailscale/tailscaled.state --socket=/run/tailscale/tailscaled.sock --port=41641

Nov 04 10:18:08 oma-vs1 tailscaled[2560749]: magicsock: endpoints changed: 24.252.3.19:33565 (stun), [2600:8804:1202:4c02:3617:ebff:fedc:470]:41641 (stun), 10.1.10.6:41641 (local)
Nov 04 10:18:08 oma-vs1 tailscaled[2560749]: magicsock: adding connection to derp-12 for home-keep-alive
Nov 04 10:18:08 oma-vs1 tailscaled[2560749]: magicsock: 1 active derp conns: derp-12=cr0s,wr0s
Nov 04 10:18:08 oma-vs1 tailscaled[2560749]: Switching ipn state Starting -> Running (WantRunning=true, nm=true)
Nov 04 10:18:08 oma-vs1 tailscaled[2560749]: derphttp.Client.Connect: connecting to derp-12 (ord)
Nov 04 10:18:08 oma-vs1 tailscaled[2560749]: control: NetInfo: NetInfo{varies=false hairpin=false ipv6=true udp=true icmpv4=false derp=#12 portmap= link=""}
Nov 04 10:18:08 oma-vs1 tailscaled[2560749]: magicsock: derp-12 connected; connGen=1
Nov 04 10:18:08 oma-vs1 tailscaled[2560749]: health("overall"): ok
Nov 04 10:18:08 oma-vs1 tailscaled[2560749]: control: netmap: got new dial plan from control
Nov 04 10:18:08 oma-vs1 tailscaled[2560749]: control: netmap: got new dial plan from control
root@oma-vs1:~# tailscale status
100.99.26.120   oma-vs1              oma-vs1.taild7e4.ts.net linux   -
```

Attempting to ping first device from second device, checking routes, still failed:

```shell
root@oma-vs1:~# tailscale status
100.99.26.120   oma-vs1              oma-vs1.taild7e4.ts.net linux   -
root@oma-vs1:~# ping 100.81.78.19 
PING 100.81.78.19 (100.81.78.19) 56(84) bytes of data.
^C
--- 100.81.78.19 ping statistics ---
6 packets transmitted, 0 received, 100% packet loss, time 5125ms

root@oma-vs1:~# ip route
default via 10.1.10.1 dev vmbr0.10 proto kernel onlink 
10.1.10.0/24 dev vmbr0.10 proto kernel scope link src 10.1.10.6 
root@oma-vs1:~# ip route add 100.0.0.0/8 dev tailscale0 
root@oma-vs1:~# ping 100.81.78.19 
PING 100.81.78.19 (100.81.78.19) 56(84) bytes of data.
^C
--- 100.81.78.19 ping statistics ---
3 packets transmitted, 0 received, 100% packet loss, time 2057ms
```

These two machines in particular sit on the same VLAN/no firewall between them (and no external firewall rules were recently changed either way).

Let me know if I can provide any additional info.  Thanks for your time.

### Steps to reproduce

- From 'Access Controls', rename a tag in ACL file (replace all instances of 'x' with 'y')
- From 'Machines', remove instances of old tag in devices' ACL tags
- From 'Machines', add new tag to same devices

I'm not sure how to verify that this was specifically what caused it, but these were the actions I took immediately before things dropped offline.

### Are there any recent changes that introduced the issue?

See above.

### OS

Linux, macOS

### OS version

Debian 11, macOS Monterrey

### Tailscale version

1.32.2

### Bug report

BUG-e23682657b1a16a48ff56435be33459278d07748d4637a391ad7fddbbf8d94c3-20221104144943Z-97f517ee66e81c4e

**Comment by DentonGentry** [comment]
The ACL for the tailnet currently contains:
```
	"acls": [
		{"action": "accept"},
	],
```

Which isn't a complete ACL, it needs a `src` and `dst` to match flows:
```
	"acls": [
		{
			"action": "accept",
			"src":  ["*"],
			"dst":  ["*:*"],
		},
	],
```

**Comment by joelheaps** [comment]
@DentonGentry Thanks for the quick reply.  You're right on the allow-all rule.  I added that to troubleshoot, but I think I also figured out what my initial problem was.

I think I had my head too far in the weeds; I might have accidentally removed the "acls: [" line when I made my initial edits 🤦.  It must be time for the weekend.

I also didn't realize that `tailscale status` would only show machines that were specifically reachable by the current machine (as opposed to just in the tailnet), so I'll keep that in mind for next time.

Closing.  Thanks for your help; sorry to bug you (pun...intended?)
